src/any_llm/any_llm.py

```
@@ -732,8 +744,17 @@ async def amessages(
             metadata=metadata,
             thinking=thinking,
             cache_control=cache_control,
+            response_format=response_format,
         )
-        return await self._amessages(params, **kwargs)
+        result = await self._amessages(params, **kwargs)
+
+        if is_structured_output_type(response_format) and isinstance(result, MessageResponse):
+            if result.output_parsed is None:
```

Ruff (0.15.15)

[warning] Use a single `if` statement instead of nested `if` statements

(SIM102)